rules-reviewed/eap7/eap6/hibernate4-xml.windup.xml

```diff
@@ -40,5 +43,42 @@
                 <matches pattern="(cfg.xml|properties)" />
             </where>
         </rule>
+        <rule id="hibernate4-00002">
+            <when>
+                <filecontent filename="{*}.hbm.xml" pattern="org.hibernate.id.{idclass}"/>
+            </when>
+            <perform>
+                <hint title="Class org.hibernate.id.{idclass} was removed/deprecated in Hibernate 5" effort="1" severity="mandatory">
+                    <message>Use `org.hibernate.id.enhanced.SequenceStyleGenerator` instead.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#deprecations"
+                        title="Removed or deprecated types in Hibernate 5" />
+                    <link
+                        href="https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/7.0.beta/migration-guide/chapter-4-application-changes#hibernate_and_jpa_migration_changes"
+                        title="Hibernate and JPA migration changes" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+            <where param="idclass">
+                <matches pattern="(TableGenerator|TableHiLoGenerator|SequenceGenerator|SequenceIdentityGenerator|SequenceHiLoGenerator)" />
+            </where>
+        </rule>
+        <rule id="hibernate4-00003">
+            <when>
+                <xmlfile matches="//generator[@class='hilo'] | //generator[@class='sequence-identity']" in="{*}.hbm.xml"/>
```

Comment:
`hilo` has indeed been removed.

`sequence-identity` is still there and corresponds to `SequenceIdentityGenerator` which is deprecated.

`seqhilo` is the same case: still present, corresponds to `SequenceHiLoGenerator`, which is also deprecated.

See:
old -- https://github.com/hibernate/hibernate-orm/blob/4.2/hibernate-core/src/main/java/org/hibernate/id/factory/internal/DefaultIdentifierGeneratorFactory.java
new -- https://github.com/hibernate/hibernate-orm/blob/5.0/hibernate-core/src/main/java/org/hibernate/id/factory/internal/DefaultIdentifierGeneratorFactory.java